src/test/java/org/jabref/logic/search/query/SearchQueryLuceneConversionTest.java

```diff
@@ -10,64 +12,55 @@
+                Arguments.of("+(content:imag annotations:process) +(content:term annotations:term)", "(content = image OR annotations = processing) AND term"),
+                Arguments.of("(content:on annotations:on) (+(content:two annotations:two) +(content:three annotations:three))", "one OR (two AND three)"),
 
-                // not converted, because wrong syntax for JabRef 5.x
-                // Arguments.of("( author:miller OR title:\"image processing\" OR keywords:\"image processing\" ) AND NOT author:brown AND NOT author:blue", "(author = miller or title|keywords = \"image processing\") and not author = brown and != author = blue"),
+                Arguments.of("(-content:term) (-annotations:term)", "any != term"),
+                Arguments.of("(-content:term) (-annotations:term)", "any !== term"),
+                Arguments.of("(-content:term) (-annotations:term)", "any !=! term"),
+                Arguments.of("(-content:\"two term\") (-annotations:\"two term\")", "any != \"two terms\""),
+                Arguments.of("+(-content:imag) +(-annotations:process)", "content != image AND annotations != processing"),
 
-                // String with a space
-                Arguments.of("title:image\\ processing AND author:smith", "title = \"image processing\" AND author= smith"),
+                Arguments.of("MatchNoDocsQuery(\"\")", "title = image"),
+                Arguments.of("content:\"imag process\" annotations:\"imag process\"", "\"image processing\" AND author = smith"),
+                Arguments.of("+(content:imag annotations:imag) +(content:process annotations:process)", "image AND (title = term OR processing)"),
+                Arguments.of("(content:imag annotations:imag) (content:process annotations:process)", "image OR (title = term OR processing)"),
+                Arguments.of("MatchNoDocsQuery(\"\")", "title = \"image processing\" AND author = smith"),
 
-                // We renamed fields
-                Arguments.of("any:somecontent", "anyfield = somecontent"),
-                Arguments.of("keywords:somekeyword", "anykeyword = somekeyword"),
-                Arguments.of("citationkey:somebibtexkey", "key = somebibtexkey")
-        );
-    }
+                Arguments.of("content:neighbou?r annotations:neighbou?r", "neighbou?r"),
+                Arguments.of("content:neighbo* annotations:neighbo*", "neighbo*"),
+                Arguments.of("MatchNoDocsQuery(\"\")", "title = neighbou?r"),
+                Arguments.of("MatchNoDocsQuery(\"\")", "(title == chocolate) OR (author == smith)"),
 
-    @ParameterizedTest
-    @MethodSource
-    void transformationNormal(String expected, String query) {
-        String result = SearchQueryConversion.searchToLucene(query);
-        assertEquals(expected, result);
-    }
-
-    public static Stream<Arguments> transformationRegularExpression() {
-        return Stream.of(
-                Arguments.of("any:* AND -groups:/.+/", "groups != .+"),
```

Comment:
Can this test be kept somehouw? I think, it checks for set groups, which is an impotant use case.

Reply:
> Can this test be kept somehouw? I think, it checks for set groups, which is an impotant use case.

I kept the test, but the query used for Lucene searches in linked files only. The query `groups != .+` will be converted to `org.apache.lucene.search.MatchNoDocsQuery`. The only searchable fields in linked files are `content` and `annotations`; any other fields will be ignored and converted to `MatchNoDocsQuery`. Queries without a specified field or using fields like `any` or `anyfield` will search in both `content` and `annotations`.